src/ansys/dpf/composites/_composite_model_impl.py

```diff
@@ -0,0 +1,613 @@
+        """Underlying DPF core model."""
+        return self._core_model
+
+    @property
+    def material_operators(self) -> MaterialOperators:
+        """Material operators."""
+        return self._material_operators
+
+    @_deprecated_composite_definition_label
+    def get_mesh(self, composite_definition_label: Optional[str] = None) -> MeshedRegion:
+        """Get the underlying DPF meshed region.
+
+        The meshed region contains the lay-up information.
+        """
+        return self._core_model.metadata.meshed_region
+
+    @_deprecated_composite_definition_label
+    def get_layup_operator(self, composite_definition_label: Optional[str] = None) -> Operator:
+        """Get the lay-up operator.
+
+        Parameters
+        ----------
+        composite_definition_label :
+            Label of the composite definition, which is the
+            dictionary key in the :attr:`.ContinuousFiberCompositesFiles.composite`
+            attribute. This parameter is only required for assemblies.
+            See the note about assemblies in the description for the :class:`CompositeModel` class.
+
+        """
+        return self._layup_provider
+
+    @_deprecated_composite_definition_label
+    def evaluate_failure_criteria(
+        self,
+        combined_criterion: CombinedFailureCriterion,
+        composite_scope: Optional[CompositeScope] = None,
+        measure: FailureMeasureEnum = FailureMeasureEnum.INVERSE_RESERVE_FACTOR,
+        write_data_for_full_element_scope: bool = True,
+        max_chunk_size: int = 50000,
+    ) -> FieldsContainer:
+        """Get a fields container with the evaluated failure criteria.
+
+        The fields container contains the maximum per element if the measure
+        is :attr:`.FailureMeasureEnum.INVERSE_RESERVE_FACTOR` and the minimum per element
+        if the measure is :attr:`.FailureMeasureEnum.MARGIN_OF_SAFETY` or
+        :attr:`.FailureMeasureEnum.RESERVE_FACTOR`.
+
+        Parameters
+        ----------
+        combined_criterion :
+            Combined failure criterion to evaluate.
+        composite_scope :
+            Composite scope on which to evaluate the failure criteria. If empty, the criteria
+            is evaluated on the full model. If the time is not set, the last time or
+            frequency in the result file is used.
+        measure :
+            Failure measure to evaluate.
+        write_data_for_full_element_scope :
+            Whether each element in the element scope is to get a
+            (potentially zero) failure value, even elements that are not
+            part of ``composite_scope.plies``. If no element scope is
+            specified (``composite_scope.elements``), a (potentially zero)
+            failure value is written for all elements.
+        max_chunk_size:
+            Maximum chunk size. If number of elements is larger than this number,
+            the failure criteria are evaluated in chunks of the specified size.
```

Comment:
```suggestion
            Maximum chunk size. The default is ``50000``. If number of elements is larger than the
            value specified, failure criterion are evaluated in chunks of the specified size.
```
You should be specifying the data type after the parameter name and whether it is optional. When it is optional, the default should be specified.

Reply:
@PipKat We decided to skip the explicit type information since chances are high that it gets outdated soon. As a developer I am looking first at the function signature including types anyway. Furthermore the types are automatically added to the sphinx docs.